kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunnerViewModel.kt

```diff
@@ -59,46 +72,67 @@ internal class WorkflowRunnerViewModel<OutputT : Any>(
       val hostFactory = WorkflowHost.Factory(dispatcher)
       val workflowHost = hostFactory.run(workflow, inputs, snapshot)
       @Suppress("UNCHECKED_CAST")
-      return WorkflowRunnerViewModel(viewRegistry, workflowHost) as T
+      return WorkflowRunnerViewModel(viewRegistry, workflowHost, dispatcher) as T
     }
   }
 
-  private val subs = CompositeDisposable()
+  private val snapshotJob = GlobalScope.launch(context) {
+    workflowHost.renderingsAndSnapshots
+        .map { it.snapshot }
+        .collect { lastSnapshot = it }
+  }
 
-  @Suppress("EXPERIMENTAL_API_USAGE")
-  private val updates =
-    workflowUpdates.updates.asObservable(Dispatchers.Unconfined)
-        .doOnNext { lastSnapshot = it.snapshot }
-        .replay(1)
-        .autoConnect(1) { subs.add(it) }
+  private var workflowJob: Job? = null
 
   private var lastSnapshot: Snapshot = Snapshot.EMPTY
 
-  override val renderings: Observable<out Any> = updates.map { it.rendering }
+  @UseExperimental(ExperimentalCoroutinesApi::class)
+  override val renderings: Observable<out Any> = workflowHost.renderingsAndSnapshots
```

Comment:
So this crafty `autoConnect()`-esque scheme means that we _don't_ have to worry about making any kind of start call around `setContentWorkflow()`, right? We still have the problem of the various races around `output()`, but that isn't made any worse by this PR.

Reply:
Correct. This gives us the same behavior we had before.